capstone/src/test/java/com/google/step/servlets/LoginServletTest.java

```diff
@@ -0,0 +1,156 @@
+        assertFalse(userService.isUserLoggedIn());
+        // Can't also check if the user is an admin because the method will throw an error since the
+        // user is not even logged in.
+    }
+
+    @Test
+    public void testDoGetWithoutLogIn() throws IOException, SAXException {
+        // Testing doGet method without having the user logged in.
+        // Result: {"LoggedIn":"false","loginURL":"someURL"}
+
+        ServletRunner sr = new ServletRunner();
+        // The first parameter has to be the same as whatever goes after the '/' in the urlString for
+        // the GetMethodWebRequest so for example if the first parameter is 'restaurant' then the link
+        // would be: 'http://localhost:8080/restaurant' even though the actual servlet may be called
+        // something else.
+        sr.registerServlet("login", LoginServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        // This can throw the SAXException.
+        WebRequest request = new GetMethodWebRequest("http://localhost:8080/login");
+        WebResponse response = sc.getResponse(request);
+        assertNotNull("No response received", response);
+        assertEquals("content type", "application/json", response.getContentType());
+
+        // Turn content to JSon
+        JsonObject jsonObject = turnStringToJson(response.getText());
+
+        assertNotNull("Return Json", response.getText());
+        assertEquals("Json Loggedin", "false", jsonObject.get("LoggedIn").toString());
+
+        // Can't check for the link to be correct, but we can check that it is not null.
+        assertNotNull("Login URL", jsonObject.get("loginURL").toString());
+    }
+
+    @Test
+    public void testDoGetWithLogIn() throws IOException, SAXException {
+        // If you set an email you must also set the auth domain.
+        helper.setEnvIsLoggedIn(true)
+                .setEnvIsAdmin(false)
+                .setEnvEmail("example@gmail.com")
+                .setEnvAuthDomain("example@gmail.com");
+
+        ServletRunner sr = new ServletRunner();
+
+        sr.registerServlet("login", LoginServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        WebRequest request = new GetMethodWebRequest("http://localhost:8080/login");
+        WebResponse response = sc.getResponse(request);
+
+        assertNotNull("No response received", response);
+        assertEquals("content type", "application/json", response.getContentType());
+
+        JsonObject jsonObject = turnStringToJson(response.getText());
+
+        assertEquals("LoggedIn", "true", jsonObject.get("LoggedIn").toString());
+
+        // Have to add the extra quotes in the expected because for some reason the JSON Parser adds
+        // them again.
+        assertEquals("Email", "\"example@gmail.com\"", jsonObject.get("Email").toString());
+
+        assertNotNull("Log out url", jsonObject.get("logOutURL").toString());
```

Comment:
Same question about if you can check the URL for correctness by constructing it with UserService?